[Fix bug 1254127] Make footer sticky to the bottom

comzeradd opened the pull request.

comzeradd commented:
@akatsoulas r?

johngian commented:
Please rebase with master.

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
 .clear{clear:both;}
 
+html, body {
+  height: 100%;
```
I don't see this getting overridden somehow. Isn't this the default value?

comzeradd replied:
Not on all browsers.

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
 body {
     #content-wrapper { background: @white; }
+    #outer-wrapper {
+        height: 100%;
```
Same here.

comzeradd replied:
On div elements the default height is usually auto (depends on the browser), not 100%.

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
+        height: 100%;
+    }
+    #page {
+        min-height: 100%;
```
Same here.

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
+        }
+    }
+    #page:after {
+        content: "";
```
How does this affect the sticky footer?

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
+        content: "";
+        display: block;
+        height: @footerDesktop;
+        background-color: @white;
```
Same here.

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
+            margin-bottom: -@footerMobile;
+        }
+    }
+    #page:after {
```
Can't we just add a placeholder element in HTML instead of adding ``:after`` content?

johngian replied:
I am not sure what's better (element vs pseudoelement) please add a comment in the code to explain why we need this.

comzeradd replied:
Yes, in general there are two ways of doing this. Either using an empty element with specific height or use an `:after` selector to the existing page wrapper. Both have the same effect. I choose the latter because our wrappers structure permits it and it allows us to keep the html code cleaner.

johngian replied:
Sounds good. Let's add a comment here to explain this for future reference.

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
+
 body {
     #content-wrapper { background: @white; }
+    #outer-wrapper {
```
Let's concat this one with the same definition above.

comzeradd commented:
PR rebased to master

johngian commented on mozillians/static/mozillians/css/main.less:
```diff
@@ -133,8 +140,38 @@
+        content: "";
+        display: block;
+        height: @footerDesktop;
+        background-color: @white;
```
let's switch the order so ``height`` is right next to the media queries that also change ``height``.

akatsoulas commented:
The profile view has a line right above the footer.
http://awesomescreenshot.com/0da5rm3lea

akatsoulas commented:
There is a line in profile_view right above the footer
![profile_view](https://cloud.githubusercontent.com/assets/900565/14281556/85390e90-fb42-11e5-81fd-5de2fa2650a6.png)

akatsoulas commented:
The main page when visiting the site seems broken

![main_page_logged_out](https://cloud.githubusercontent.com/assets/900565/14281601/db6bfae8-fb42-11e5-936d-8e479ca8f428.png)

comzeradd commented:
@akatsoulas  I fixed the first issue. The problem was that we have the sandstone background to the whole website, instead of just the header where we actually use it.

The second problem is caused due to the fact that the "Get Involved" section is not sticky on its own. In order to stick it to the bottom we should move outside the `#page` wrapper and chage heights and margins specifically for this case (home page and not logged in).

That significantly increases the complexity of the code, but I agree it looks better. I added a second commit for that to see what I mean. I'll squash it if we decide to go this way.

johngian commented on mozillians/jinja2/base.html:
```diff
@@ -179,6 +179,8 @@
 
+  {% block get_involved %}{% endblock %}
+
     {% block site_footer %}
```
Please fix identation.

johngian commented:
Logged out page background is broken. (ignore the django error)
![selection_024](https://cloud.githubusercontent.com/assets/835321/14486478/d393b044-0165-11e6-94f4-0f1bc39f3fcb.png)
![selection_025](https://cloud.githubusercontent.com/assets/835321/14486495/e9138c3c-0165-11e6-9f0d-9a89fe80790b.png)

comzeradd commented:
PR updated. Indentation was off all over the base template.

@johngian I'm having hard time to determine the problem at your last screenshots. Can you post a full page one.

johngian commented:
![selection_026](https://cloud.githubusercontent.com/assets/835321/14522439/304df412-0237-11e6-8d23-269fe88ace8a.png)
![selection_027](https://cloud.githubusercontent.com/assets/835321/14522474/50a4a45e-0237-11e6-8cf0-e4273a272db8.png)

comzeradd commented:
PR updated with background color set back to sandstone for logged out home page.

ppapadeas commented:
@comzeradd when logged out, navigating to a profile now displays wrong color for filler. See image
![screen shot 2016-04-16 at 13 17 36](https://cloud.githubusercontent.com/assets/498592/14580689/b37bc554-03d5-11e6-95cb-43ff135cc923.png)'

Presumably this happened with the final fix for logged out main page. It should be filling with main color of the page, not hardcoded.

comzeradd commented:
@ppapadeas good catch. PR updated. I fixed the `#page` background, since the loggedout homepage is the only place we use something else instead of white.

akatsoulas commented:
@johngian could you please have another pass at this?

johngian commented:
Please rebase

johngian commented on mozillians/jinja2/base.html:
```diff
@@ -57,127 +57,129 @@
+            </ul>
+            {% block site_header_nav %}
+              <nav role="navigation">
+                  <a class="brand" href="{{ url('phonebook:home') }}">
```
Since you are fixing the indentation of this file please fix that one as well.

johngian commented on mozillians/jinja2/phonebook/home.html:
```diff
@@ -274,6 +254,30 @@
 
 {% endblock content %}
 
+{% block get_involved %}
```
Since we are using this ``<article>`` only for non-authenticated user's home page, why are you moving it to it's own block that also adds the complexity to check if user is authenticated?

akatsoulas replied:
If this is not needed for the sticky footer, please revert this change.

comzeradd replied:
The "get involved" should be moved between `#page` and `footer`, thus the change. It can't live inside the content block.

johngian replied:
sounds good.

comzeradd commented:
PR rebased. I also fixed indentation issues on base template.

akatsoulas commented:
Tested this locally and looks good. r+ from me

johngian commented:
On mobile view (360x640) ``#get_involved.content`` is misaligned to the left. It's on logged out home page.

johngian commented:
r+wc from me

comzeradd commented:
> On mobile view (360x640) #get_involved.content is misaligned to the left. It's on logged out home page.

I added an explicit `margin-left` to `#get_involved.content` to make sure it's rendered properly on mobile views.
